### PR TITLE
fix: restored compatibility with changed return structures

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-243

- [ ] https://github.com/oat-sa/extension-tao-dac-simple/pull/159

### CODE REVIEW

File: models/classes/search/SearchProxy.php [hunk 1]
```diff
@@ -15,26 +15,43 @@
...
 use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\tao\model\search\strategy\GenerisSearch;
 use Psr\Http\Message\ServerRequestInterface;
 
 class SearchProxy extends ConfigurableService
 {
     use OntologyAwareTrait;
 
+    private const GENERIS_SEARCH_WHITELIST = [
+        GenerisRdf::CLASS_ROLE
```
Comment (gabrielfs7) [inline models/classes/search/SearchProxy.php:38]: Isn't missing the new entry here for the User URI?

File: models/classes/search/SearchProxy.php [hunk 2]
```diff
@@ -15,26 +15,45 @@
...
     use OntologyAwareTrait;
 
+    private const GENERIS_SEARCH_WHITELIST = [
+        GenerisRdf::CLASS_ROLE,
+        TaoOntology::CLASS_URI_TAO_USER,
+    ];
+
+    /**
+     * @throws Exception
+     */
```
Comment (shpran) [inline models/classes/search/SearchProxy.php:45]: We don't need this comment

File: models/classes/search/SearchProxy.php [hunk 3]
```diff
@@ -85,4 +108,25 @@ private function getQueryFactory(): SearchQueryFactory
     {
         return $this->getServiceLocator()->get(SearchQueryFactory::class);
     }
+
+    private function isForcingGenerisSearch(SearchQuery $query): bool
+    {
+        return in_array($query->getParentClass(), self::GENERIS_SEARCH_WHITELIST);
```
Comment (shpran) [inline models/classes/search/SearchProxy.php:114]:
```suggestion
        return in_array($query->getParentClass(), self::GENERIS_SEARCH_WHITELIST, true);
```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: models/classes/search/SearchProxy.php (hunk 2 above)
Comment (boajer) [reply models/classes/search/SearchProxy.php:45]: Why?

**Review by pnal — APPROVED** [review]
Looks OK
upd: do not forget to fix unit test which failing after last commit
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by gabrielfs7 — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**